pkgs/by-name/fl/flattenReferencesGraph/src/flatten_references_graph/lib.py

```diff
@@ -0,0 +1,329 @@
+from collections.abc import Iterable
+from pathlib import Path
+from toolz import curried as tlz
+from toolz import curry
+import igraph as igraph
+import itertools as itertools
+import json as json
+import os as os
+import re as re
+import sys
+
+DEBUG = os.environ.get("DEBUG", False) == "True"
+DEBUG_PLOT = os.environ.get("DEBUG_PLOT", False) == "True"
+# If this is set, the plots will be saved to files instead of being displayed
+# with default image viewer.
+DEBUG_PLOT_SAVE_BASE_NAME = os.environ.get("DEBUG_PLOT_SAVE_BASE_NAME")
+
+c = igraph.configuration.init()
+# App used to open the plots when DEBUG_PLOT_SAVE_BASE_NAME is not set.
+c["apps.image_viewer"] = os.environ.get("DEBUG_PLOT_IMAGE_VIEWER", "gwenview")
+
+
+def debug(*args, **kwargs):
+    if DEBUG:
+        print(*args, file=sys.stderr, **kwargs)
+
+
+def debug_plot(graph, name, **kwargs):
+    if not DEBUG_PLOT:
+        return
+
+    vertex_label = [
+        # remove /nix/store/HASH- prefix from labels
+        re.split("^/nix/store/[a-z0-9]{32}-", name)[-1]
+        for name in graph.vs["name"]
+    ]
+
+    save_as = (
+        None if DEBUG_PLOT_SAVE_BASE_NAME is None
+        else DEBUG_PLOT_SAVE_BASE_NAME + name + ".png"
+    )
+
+    igraph.plot(
+        graph,
+        save_as,
+        vertex_label=vertex_label,
+        **(tlz.merge(
+            {
+                # "bbox": (3840, 2160),
+                "bbox": (800, 600),
+                "margin": 100,
+                "vertex_label_dist": -5,
+                "edge_color": "orange",
+                "vertex_size": 20,
+                "vertex_label_size": 30,
+                "edge_arrow_size": 2
+            },
+            kwargs
+        )),
+    )
+
+
+def debug_plot_with_highligth(g, vs, layout):
+    debug_plot(
+        g,
+        layout=layout,
+        # layout=Layout(new_coords),
+        vertex_color=[
+            "green" if v.index in vs else "red"
+            for v in g.vs
+        ]
+    )
+
+
+@curry
+def pick_keys(keys, d):
+    return {
+        key: d[key] for key in keys if key in d
+    }
+
+
+def unnest_iterable(xs):
+    return itertools.chain.from_iterable(xs)
+
+
+def load_json(file_path):
+    with open(file_path) as f:
+        return json.load(f)
+
+
+@curry
+def sorted_by(key, xs):
+    return sorted(xs, key=lambda x: x[key])
+
+
+@curry
+def find_vertex_by_name_or_none(graph, name):
+    try:
+        # NOTE: find by name is constant time.
+        return graph.vs.find(name)
+    # This will be thrown if vertex with given name is not found.
+    except ValueError:
+        return None
+
+
+def subcomponent_multi(graph, vertices, mode="out"):
+    """Return concatenated subcomponents generated by the given list of
+    vertices.
+    """
+    return tlz.mapcat(
+        lambda vertex: graph.subcomponent(vertex, mode=mode),
+        vertices
+    )
+
+
+@curry
+def egdes_for_reference_graph_node(path_to_size_dict, reference_graph_node):
```

Comment:
```suggestion
def edges_for_reference_graph_node(path_to_size_dict, reference_graph_node):
```
plus places where it gets called.

Reply:
fixed in 88bdfb156538, thanks!